res/controllers/Yaeltex-MiniMixxx-scripts.js

```diff
@@ -0,0 +1,1511 @@
+///////////////////////////////////////////////////////////////////////////////////
+//
+// MiniMixxx controller script v1.00
+// Author: Owen Williams (owilliams@mixxx.org)
+//
+///////////////////////////////////////////////////////////////////////////////////
+//
+// TODO:
+//
+///////////////////////////////////////////////////////////////////////////////////
+
+var MiniMixxx = {};
+
+MiniMixxx.FXColor = 61;           // Cyan
+MiniMixxx.LoopColor = 46;         // Green
+MiniMixxx.LibraryColor = 104;     // Purple
+MiniMixxx.LibraryFocusColor = 97; // More Purple
+MiniMixxx.GainColor = 96;         // Pale Blue
+MiniMixxx.JogColor = 13;          // Orange
+MiniMixxx.PeakColor = 1;          // Red
+MiniMixxx.HotcueColor = 20;       // Light Orange
+MiniMixxx.UnsetSamplerColor = 82; // Blue
+MiniMixxx.SamplerColor = 70;      // Light Blue
+MiniMixxx.MainOutColor = 50;      // Kind of a light green
+MiniMixxx.HeadOutColor = 26;      // Kind of a light yellow
+
+// Set to true to output debug messages and debug light outputs.
+MiniMixxx.DebugMode = false;
+
+// Set to true to only allow adjustments to the pitch sliders if Shift is held.
+// This can prevent accidental adjustments.
+MiniMixxx.ShiftPitch = false;
+
+// An Encoder represents a single encoder knob and tracks the active mode.
+MiniMixxx.Encoder = class {
+    constructor(channel, idx, layerConfig) {
+        this.channel = channel;
+        this.idx = idx;
+        this.encoders = {};
+        this.layers = {};
+        this.activeMode = {};
+
+        for (const layerName in layerConfig) {
+            const mode = layerConfig[layerName];
+            if (mode === "JOG") {
```

Comment:
Why not use an enum and a switch statement instead of string comparison?

Reply:
oh hah I didn't know js could do that.  I'll fix

Reply:
done